tests/test_routedem.py

```diff
@@ -259,10 +264,39 @@ def test_routedem_d8(self):
         numpy.testing.assert_allclose(
             expected_downslope_distance,
             gdal.OpenEx(os.path.join(
-                args['workspace_dir'], 
+                args['workspace_dir'],
                 'downslope_distance_foo.tif')).ReadAsArray(),
             rtol=0, atol=1e-6)
 
+        try:
+            vector = gdal.OpenEx(os.path.join(
+                args['workspace_dir'], 'strahler_stream_order_foo.gpkg'))
+            layer = vector.GetLayer()
+            self.assertEqual(27, layer.GetFeatureCount())
+            features_per_order = collections.defaultdict(int)
+            for feature in layer:
+                order = feature.GetField('order')
+                features_per_order[order] += 1
+            self.assertEqual(dict(features_per_order), {1: 18, 2: 9})
+        finally:
+            layer = None
+            vector = None

```

Comment:
This is neat, even if something goes awry we should still try and cleanup. I like that.